templates.tf

```diff
@@ -19,7 +32,8 @@ data "template_file" "server_yaml" {
   template = file("${path.module}/templates/waggle-dance-server.yml.tmpl")
 
   vars = {
-    graphite = join("", data.template_file.graphite_server_yaml.*.rendered)
+    graphite          = join("", data.template_file.graphite_server_yaml.*.rendered)
+    exposed_endpoints = join("", data.template_file.endpoints_server_yaml.rendered)
```

Comment:
may not need join here, join was mainly use to make blocks optional in terraform 11, seems like we will be configuring default enpoint list when prometheus is not enabled.
exposed_endpoints = join("", data.template_file.endpoints_server_yaml.rendered)

so we can just setup as
exposed_endpoints = data.template_file.endpoints_server_yaml.rendered

also FYI we can refactor waggle-dance server yaml using if conditions in a single file in terraform 12, I will create another PR for that.

Reply:
Make sense. Do you have any idea if there are any other endpoints in use, aside from the one in the default list here?